src/components/LinkEditor.tsx

```diff
@@ -35,14 +50,19 @@ type State = {
 
 class LinkEditor extends React.Component<Props, State> {
   discardInputValue = false;
-  initialValue: string = this.props.mark.attrs.href;
+  initialValue = this.href;
+  hasInitialSelection = this.props.from !== this.props.to;
```

Comment:
A collapsed selection is still a selection, would maybe suggest flipping this to:

```suggestion
  initialSelectionCollapsed = this.props.from === this.props.to;
```

Reply:
That sounds fine, are you going to change it to a number instead of a boolean then?